staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/clientset/versioned/typed/cr/v1/cr_client.go

```diff
@@ -134,7 +138,8 @@ func (c *CrV1Client) RESTClients() []rest.Interface {
 	if c == nil {
 		return nil
 	}
-
+	c.mux.RLock()
+	defer c.mux.RUnlock()
```

Comment:
Not sure whether this particular one helps

Reply:
understood your point( actually i was in the same line when i added it as well). they were there just for ensuring the function was locked for all getters.

Reply:
defer actually takes sometime to finish. So I think this one would cause unnecessary delay.

Reply:
i cannot find perf hit data in golang.org but some other [blog](https://kylewbanks.com/blog/when-to-use-defer-in-go) claims the perf impact of defer in golang is nanosecond level so it is ok in terms of perf concern here. in fact, in all k8s code, defer is commonly used for all different purposes.

Reply:
will resolve for now. we can always do micro-optimization later on.

Reply:
removed the lock for the return statement in this Getter. per Ying's suggestion this is heavily used getter so perf might be concerned here.